### PR TITLE
Add method to limit image sizes based on dimensions

### PR DESCRIPTION
**Pull request by lgaleana-llm**
### Summary

- Added environment variables for minimum image dimensions.
- Updated the `extract_images` function to filter images based on dimensions.
- Added unit tests to verify the new functionality.

### Details

1. **Updated Environment Variables:**
   - Added `MIN_IMAGE_WIDTH` and `MIN_IMAGE_HEIGHT` to `.env.example` to specify the minimum width and height for images.

2. **Modified `extract_images` Function:**
   - Updated `extract_images` in `app/scraping.py` to filter out images that do not meet the minimum size requirements.

3. **Added Unit Tests:**
   - Created `tests/test_image_size_filter.py` to test the new functionality.

### Test Plan

pytest / playwright

### CODE REVIEW

**Comment by lgaleana-llm** [comment]
!!!!! WARNING: Unrelated tests aren't passing !!!!!

**Comment by lgaleana-llm** [comment]
Total time to create PR: 118.37 seconds
Total OPenAI cost: 0.59398

File: app/scraping.py [hunk 1]
```diff
@@ -1,19 +1,42 @@
...
 import os
+import requests
+from PIL import Image
+from io import BytesIO
 
 
 def extract_images(soup: BeautifulSoup) -> list:
     images = []
     n_images = int(os.getenv("N_IMAGES", 4))  # Default to 4 if not set
+    min_width = int(os.getenv("MIN_IMAGE_WIDTH", 0))
```
Comment (lgaleana) [inline app/scraping.py]: Let's hardcode this for now

Comment (lgaleana-llm) [reply app/scraping.py]: Sure, I can hardcode the minimum width and height for now. I'll update the code accordingly. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
!!!!! WARNING: Unrelated tests aren't passing !!!!!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 59.16 seconds
Total OPenAI cost: 0.35623000000000005

File: app/scraping.py [hunk 2]
```diff
@@ -1,19 +1,41 @@
...
 import os
-
+import requests
+from PIL import Image
+from io import BytesIO
 
 def extract_images(soup: BeautifulSoup) -> list:
     images = []
     n_images = int(os.getenv("N_IMAGES", 4))  # Default to 4 if not set
+    min_width = 100  # Hardcoded minimum width
```
Comment (lgaleana) [inline app/scraping.py]: Put this as a global variable

File: .env.example [hunk 3]
```diff
@@ -1,2 +1,4 @@
 OPENAI_API_KEY=your_openai_api_key_here
 N_IMAGES=5
+MIN_IMAGE_WIDTH=100
```
Comment (lgaleana) [inline .env.example]: Remove this

File: app/scraping.py (hunk 2 above)
Comment (lgaleana-llm) [reply app/scraping.py]: Sure, I can move the hardcoded minimum width and height values to global variables. I'll update the code accordingly. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
!!!!! WARNING: Unrelated tests aren't passing !!!!!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 99.45 seconds
Total OPenAI cost: 0.30807500000000004

File: .env.example (hunk 3 above)
Comment (lgaleana-llm) [reply .env.example]: Sure, I will remove the `MIN_IMAGE_WIDTH` and `MIN_IMAGE_HEIGHT` from the `.env.example` file. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 116.68 seconds
Total OPenAI cost: 0.3133

File: app/scraping.py [hunk 4]
```diff
@@ -1,19 +1,43 @@
 from bs4 import BeautifulSoup
 import os
+import requests
+from PIL import Image
+from io import BytesIO
 
+# Global variables for minimum width and height
+MIN_IMAGE_WIDTH = 100
```
Comment (lgaleana) [inline app/scraping.py]: Make these 300 x 250

Comment (lgaleana-llm) [reply app/scraping.py]: Sure, I will update the global variables to have a minimum width of 300 and a minimum height of 250. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 161.72 seconds
Total OPenAI cost: 0.43056

File: app/scraping.py [hunk 5]
```diff
@@ -1,19 +1,43 @@
...
+            image_url = img["srcset"].split()[0]  # Take the first URL in srcset
+        else:
+            continue
+
+        # Check image dimensions
+        try:
+            response = requests.get(image_url)
+            response.raise_for_status()
+            image = Image.open(BytesIO(response.content))
+            width, height = image.size
```
Comment (lgaleana) [inline app/scraping.py:40]: Add some logs of image names and sizes

Comment (lgaleana-llm) [reply app/scraping.py:40]: Sure, I will add some logging to display the image URLs and their sizes. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 193.39 seconds
Total OPenAI cost: 0.37451500000000004

File: app/scraping.py [hunk 6]
```diff
@@ -1,19 +1,48 @@
...
+            logging.info(f"Image URL: {image_url}, Width: {width}, Height: {height}")
+            if width >= MIN_IMAGE_WIDTH and height >= MIN_IMAGE_HEIGHT:
+                images.append(image_url)
+        except requests.RequestException:
```
Comment (lgaleana) [inline app/scraping.py]: Add some logs for exceptions too

Comment (lgaleana-llm) [reply app/scraping.py]: Sure, I will add logging for exceptions as well. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 223.76 seconds
Total OPenAI cost: 0.4049

File: app/scraping.py [hunk 7]
```diff
@@ -1,19 +1,50 @@
...
+            image_url = img["href"]
         elif img.name == "source" and img.get("srcset"):
-            images.append(img["srcset"].split()[0])  # Take the first URL in srcset
+            image_url = img["srcset"].split()[0]  # Take the first URL in srcset
+        else:
+            continue
+
+        # Check image dimensions
+        try:
+            response = requests.get(image_url)
```
Comment (lgaleana) [inline app/scraping.py]: I'm seeing a lot of errors like ERROR:root:RequestException for URL https://www.tob-dev.com/wp-content/uploads/2017/11/travel-to-bermuda.png: 406 Client Error: Not Acceptable for url: https://www.tob-dev.com/wp-content/uploads/2017/11/travel-to-bermuda.png

Comment (lgaleana-llm) [reply app/scraping.py]: I will add a user-agent header to the requests to address the 406 errors. Thank you for the feedback!

**Comment by lgaleana-llm** [comment]
Total time to amend PR: 261.99 seconds
Total OPenAI cost: 0.43718

**Review by lgaleana — APPROVED** [review]

**Pull request merged**